src/main/java/org/jitsi/jicofo/rest/Health.java

```diff
@@ -271,4 +291,126 @@ private static int cacheStatus(int status)
+                StringBuilder dbg = new StringBuilder();
+                for (ThreadInfo threadInfo : threadInfos)
+                {
+                    dbg.append('"')
+                        .append(threadInfo.getThreadName()).append('"');
+
+                    Thread.State state = threadInfo.getThreadState();
+                    dbg.append("\n   java.lang.Thread.State: ")
+                        .append(state);
+
+                    if (threadInfo.getLockName() != null)
+                    {
+                        dbg.append(" on ").append(threadInfo.getLockName());
+                    }
+                    dbg.append('\n');
+
+                    StackTraceElement[] stackTraceElements
+                        = threadInfo.getStackTrace();
+                    for (int i = 0; i < stackTraceElements.length; i++)
+                    {
+                        StackTraceElement ste = stackTraceElements[i];
+                        dbg.append("\tat " + ste.toString());
+                        dbg.append('\n');
+                        if (i == 0 && threadInfo.getLockInfo() != null)
+                        {
+                            Thread.State ts = threadInfo.getThreadState();
+                            if (ts == Thread.State.BLOCKED
+                                || ts == Thread.State.WAITING
+                                || ts == Thread.State.TIMED_WAITING)
+                            {
+                                dbg.append("\t-  " + ts + " on "
+                                    + threadInfo.getLockInfo());
+                                dbg.append('\n');
+                            }
+                        }
+
+                        for (MonitorInfo mi
+                                : threadInfo.getLockedMonitors())
+                        {
+                            if (mi.getLockedStackDepth() == i) {
+                                dbg.append("\t-  locked " + mi);
+                                dbg.append('\n');
+                            }
+                        }
+                    }
+                    dbg.append("\n\n");
+                }
+                logger.error("Health check took "
+                    + (System.currentTimeMillis() - this.startedAt)
+                    + " ms. \n"
+                    + dbg.toString());
+            }
+            catch (Throwable t)
+            {
+                logger.error(
+                    "Error printing debug info for late health checks", t);
+            }
+        }
+
+        /**
+         * Starts the monitor. Schedules execution in separate thread
+         * after some interval, if monitor is not stopped and it executes
+         * we consider the health check took too much time.
+         */
+        public void start()
+        {
+            this.startedAt = System.currentTimeMillis();
+            this.monitorTask = FocusBundleActivator.getSharedThreadPool()
```

Comment:
I was thinking that maybe it would be better to use it's own Timer for that task. It's because in case the shared scheduled executor is blocked with many scheduled tasks then it may not execute the monitor. We should not create more than 1 health task at a time, so too many threads should not be a concern here.